File: frontend/src/types/patient.ts
export type VitalParameter =
  | "temperature"
  | "blood_pressure"
  | "heart_rate"
  | "oxygen_saturation"
  | "respiratory_rate"
  | "pain_score";

export interface Allergy {
  substance: string;
  reaction?: string;
  severity: "mild" | "moderate" | "severe";
}

export interface Medication {
  name: string;
  dose?: string;
  frequency?: string;
  route?: string;
  status: "active" | "paused" | "discontinued";
  started_at?: string;
}

export interface VitalReading {
  parameter: VitalParameter;
  value: number;
  unit: string;
  recorded_at: string;
}

export interface OpenDiagnostic {
  name: string;
  ordered_at?: string;
  status: "pending" | "in_progress" | "resulted";
  result?: string;
}

export interface PatientDemographics {
  age: number;
  sex: "M" | "F" | "D";
  name: string;
}

export interface Patient {
  patient_id: string;
  bed: string;
  demographics: PatientDemographics;
  main_diagnosis: string;
  allergies: Allergy[];
  medications: Medication[];
  recent_vitals: VitalReading[];
  open_diagnostics: OpenDiagnostic[];
}

export interface PatientSummary {
  patient_id: string;
  bed: string;
  demographics: PatientDemographics;
  main_diagnosis: string;
}

File: backend/app/models/patient.py
from pydantic import BaseModel, Field
from typing import Literal


class Allergy(BaseModel):
    substance: str
    reaction: str | None = None
    severity: Literal["mild", "moderate", "severe"] = "moderate"


class Medication(BaseModel):
    name: str
    dose: str | None = None
    frequency: str | None = None
    route: str | None = None
    status: Literal["active", "paused", "discontinued"] = "active"
    started_at: str | None = None  # ISO date


class VitalReading(BaseModel):
    parameter: Literal[
        "temperature",
        "blood_pressure",
        "heart_rate",
        "oxygen_saturation",
        "respiratory_rate",
        "pain_score",
    ]
    value: float
    unit: str
    recorded_at: str  # ISO datetime


class OpenDiagnostic(BaseModel):
    name: str
    ordered_at: str | None = None  # ISO datetime
    status: Literal["pending", "in_progress", "resulted"] = "pending"
    result: str | None = None


class PatientDemographics(BaseModel):
    age: int
    sex: Literal["M", "F", "D"]
    name: str


class Patient(BaseModel):
    patient_id: str
    bed: str
    demographics: PatientDemographics
    main_diagnosis: str
    allergies: list[Allergy] = Field(default_factory=list)
    medications: list[Medication] = Field(default_factory=list)
    recent_vitals: list[VitalReading] = Field(default_factory=list)
    open_diagnostics: list[OpenDiagnostic] = Field(default_factory=list)


class PatientSummary(BaseModel):
    """Lightweight patient object returned by GET /api/patients."""

    patient_id: str
    bed: str
    demographics: PatientDemographics
    main_diagnosis: str
